ksqldb-common/src/main/java/io/confluent/ksql/util/KsqlConfig.java

```diff
@@ -1684,14 +1718,18 @@ private KsqlConfig(final ConfigGeneration generation,
     this.ksqlStreamConfigProps = ksqlStreamConfigProps;
   }
 
-  public Map<String, Object> getKsqlStreamConfigProps(final String applicationId) {
+  public Map<String, Object> getKsqlStreamConfigProps(final String applicationId, boolean addConfluentMetricsContextConfigsKafka) {
```

Comment:
Rather than adding the boolean flag here, why not just remove the code from this function and make the caller explicitly call `addConfluentMetricsContextConfigsKafka()` after this method returns?  The `addConfluentMetricsContextConfigsKafka()` method is public anyway.

This may also reduce blast radius of this change since you wouldn't need to change the signature on the mocks in the unit tests.

I realize this returns `unmodifiableMap` but you can work around that.